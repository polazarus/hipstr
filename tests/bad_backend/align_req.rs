use core::mem::size_of;

use hipstr::Backend;

struct BadBackend;

unsafe impl Backend for BadBackend {
    type RawPointer = [u8; size_of::<usize>()];

    #[inline]
    fn new(_: Vec<u8>) -> Self {
        panic!()
    }

    #[inline]
    fn try_unwrap(self) -> Result<Vec<u8>, Self> {
        panic!()
    }

    #[inline]
    fn into_raw(self) -> Self::RawPointer {
        panic!()
    }

    #[inline]
    unsafe fn from_raw(_raw: Self::RawPointer) -> Self {
        panic!()
    }

    #[inline]
    unsafe fn raw_increment_count(_raw: Self::RawPointer) {
        panic!()
    }

    #[inline]
    unsafe fn raw_decrement_count(_: Self::RawPointer) {
        panic!()
    }

    #[inline]
    unsafe fn raw_is_unique(_: Self::RawPointer) -> bool {
        panic!()
    }

    #[inline]
    unsafe fn raw_as_vec<'a>(_: Self::RawPointer) -> &'a Vec<u8> {
        panic!()
    }

    #[inline]
    unsafe fn raw_get_mut_unchecked<'a>(_: Self::RawPointer) -> &'a mut Vec<u8> {
        panic!()
    }

    #[inline]
    fn raw_is_valid(_: Self::RawPointer) -> bool {
        panic!()
    }
}

const _: () = {
    hipstr::check_backend::<BadBackend>();
};

fn main() {
    let _h = hipstr::string::HipStr::<'static, BadBackend>::new();
}
