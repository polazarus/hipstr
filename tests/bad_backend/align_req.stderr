error[E0080]: evaluation of constant value failed
  --> src/backend.rs
   |
   |         assert!(align_of::<B::RawPointer>() > 1);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: align_of::<B::RawPointer>() > 1', $DIR/src/backend.rs:134:9
   |
note: inside `check_backend::<BadBackend>`
  --> src/backend.rs
   |
   |         assert!(align_of::<B::RawPointer>() > 1);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `_`
  --> tests/bad_backend/align_req.rs:62:5
   |
62 |     hipstr::check_backend::<BadBackend>();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)
